### DISCUSSION TITLE
PYTHONPATH in .env file ignored with pyright extraPaths config

### DISCUSSION THREAD

**Discussion by propertone** [microsoft/pylance-release#2916] [Enhancement]
## Environment data

-   Language Server version: 2022.1.1
-   OS and version: Linux RedHat 7
-   Python version: 3.9.7

## Issue

This is a follow-up to #2195 - we want to merge the project's and user's `extraPath` setting. To do that, we have the `extraPath` variable specified inside a `pyproject.toml` file using:
```
[tool.pyright]
extraPaths = ["src/python"]
```

I also have a per-user PYTHONPATH specified inside a `.env` file like:
```
PYTHONPATH=/path/to/additional/libs
```
I expect that the paths in `pyproject.toml` and `.env` file will get merged, but the PYTHONPATH from the `.env` file is completely ignored instead. Using `pyrightconfig.json` has the same behavior. Merging the paths works if the extraPaths is specified ONLY in the vscode setting's `python.analysis.extraPaths` property (and NOT in the pyproject.toml/pyrightconfig.json file). However we prefer to not use this approach because: 1) Other pyright configurations like `include` are only supported in the pyright configuration file and keeping all pyright configurations in one place is more convenient, and 2) we prefer not to commit the vscode setting file to Git as some user actions unknowingly end up modifying the project's .vscode/settings.json file which causes conflicts.

**Comment by judej** [comment]
Thanks for the report. Will look  at how this should work

**Comment by heejaechang** [comment]
we need to do expanding env:PYTHONPATH in other configurations as well.

see https://github.com/microsoft/pyrx/pull/852 and https://github.com/microsoft/vscode-python/pull/15107

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
